core/base-service/service-definitions.js

```diff
@@ -46,6 +46,28 @@ const serviceDefinition = Joi.object({
       })
     )
     .default([]),
+  openApi: Joi.object().pattern(
```

Comment:
This is not indented to be a full validation of every allowable case in the spec - just a subset that is relevant for shields.

This might need a bit of tweaking as we add more cases